src/AgentCommerce/HoneyWebhookService.php

```diff
@@ -0,0 +1,175 @@
+use Lcobucci\JWT\Encoding\JoseEncoder;
+use Lcobucci\JWT\Signer\Hmac\Sha256;
+use Lcobucci\JWT\Signer\Key\InMemory;
+use Lcobucci\JWT\Token\Builder;
+use Psr\Log\LoggerInterface;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\Country\CountryEntity;
+use Shopware\Core\System\SalesChannel\SalesChannelCollection;
+use Shopware\Core\System\SalesChannel\SalesChannelEntity;
+use Shopware\Core\System\SystemConfig\SystemConfigService;
+use Swag\PayPal\AgentCommerce\Exception\HoneyWebhookExceptions;
+use Swag\PayPal\Setting\Service\CredentialsUtil;
+use Swag\PayPal\Setting\Settings;
+use Swag\PayPal\SwagPayPal;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class HoneyWebhookService
+{
+    /**
+     * @param EntityRepository<SalesChannelCollection> $salesChannelRepository
+     */
+    public function __construct(
+        private readonly ClientInterface $client,
+        private readonly EntityRepository $salesChannelRepository,
+        private readonly CredentialsUtil $credentialsUtil,
+        private readonly RouterInterface $router,
+        private readonly SystemConfigService $systemConfigService,
+        private readonly LoggerInterface $logger,
+    ) {
+    }
+
+    public function register(string $salesChannelId, Context $context): HoneyWebhookResult
+    {
+        try {
+            $salesChannel = $this->loadSalesChannel($salesChannelId, $context);
+            if (!$salesChannel || !$salesChannel->getActive()) {
+                throw HoneyWebhookExceptions::invalidSalesChannel();
+            }
+
+            $oldToken = $this->systemConfigService->get(Settings::AGENT_COMMERCE_ONBOARDED, $salesChannelId);
+            if (\is_string($oldToken)) {
+                $deregisterResult = $this->webhookCall($oldToken, 'uninstall');
+
+                if (!$deregisterResult->success) {
+                    throw HoneyWebhookExceptions::failedDeregisterWebhook();
+                }
+            }
+
+            $token = $this->createToken($salesChannel);
+        } catch (HoneyWebhookExceptions $e) {
+            $this->logger->error('PayPal agent commerce webhook livecycle: {message}', ['message' => $e->getMessage(), 'exception' => $e]);
+
+            throw $e;
+        }
+
+        $result = $this->webhookCall($token, 'install');
+        if ($result->success) {
+            $this->systemConfigService->set(Settings::AGENT_COMMERCE_ONBOARDED, $token, $salesChannelId);
+        } else {
+            $this->systemConfigService->delete(Settings::AGENT_COMMERCE_ONBOARDED, $salesChannelId);
+
+            throw HoneyWebhookExceptions::invalidRequest($result->exception);
+        }
+
+        return $result;
```

Comment:
```suggestion
        try {
            $result = $this->webhookCall($token, 'install');
            $this->systemConfigService->set(Settings::AGENT_COMMERCE_ONBOARDED, $token, $salesChannelId);
            return $result;
        } catch (HoneyWebhookExceptions $e) {
             $this->logger->error('PayPal agent commerce webhook register: {message}', ['message' => $e->getMessage(), 'exception' => $e]);
             if (\str_starts_with($e->getErrorCode(), HoneyWebhookExceptions::API_ERROR)) {
                $this->systemConfigService->delete(Settings::AGENT_COMMERCE_ONBOARDED, $salesChannelId);
            }
            
            throw $e
        }
```

Reply:
if you call the `deregister` function, you shouldn't need a second one 🤔

Reply:
but I'm also fine if you don't throw an exception in `webhookCall`, as long as the `invalidRequest` exception inherits the error code thrown by the API